new_reduce.rb

```diff
@@ -0,0 +1,9 @@
+class Array
+  def my_reduce(joint)
+    if block_given?
+      self.each do |i|
+        # something clever
```

Comment:
To be more clear: the block accesses parameters that `Array#my_reduce` passes to it.